spek-junit-platform-engine2/src/main/kotlin/org/spekframework/junit/RuntimeExecutionListenerAdapter.kt

```diff
@@ -0,0 +1,64 @@
+package org.spekframework.junit
+
+import org.junit.platform.engine.EngineExecutionListener
+import org.junit.platform.engine.TestExecutionResult
+import org.spekframework.runtime.execution.ExecutionContext
+import org.spekframework.runtime.execution.ExecutionResult
+import org.spekframework.runtime.execution.RuntimeExecutionListener
+import org.spekframework.runtime.scope.ActionScopeImpl
+import org.spekframework.runtime.scope.GroupScopeImpl
+import org.spekframework.runtime.scope.ScopeImpl
+import org.spekframework.runtime.scope.TestScopeImpl
+
+class RuntimeExecutionListenerAdapter(val listener: EngineExecutionListener): RuntimeExecutionListener() {
+    fun ScopeImpl.asDescriptor() = TestDescriptorAdapter.asDescriptor(this)
+    fun ExecutionResult.toJunitResult() = when (this) {
+        is ExecutionResult.Success -> TestExecutionResult.successful()
+        is ExecutionResult.Failure -> TestExecutionResult.failed(this.cause)
+    }
+
+    override fun executionStart() { }
+
+    override fun executionFinish() { }
+
+    override fun testExecutionStart(test: TestScopeImpl) {
+        listener.executionStarted(test.asDescriptor())
+    }
+
+    override fun testExecutionFinish(test: TestScopeImpl, result: ExecutionResult) {
+        listener.executionFinished(test.asDescriptor(), result.toJunitResult())
```

Comment:
Can we allow listener to fail test in this callback? That would be very handy for mimicing JUnit Rules

Reply:
Is there already that different PR ?